strongbox-storage/strongbox-storage-core/src/main/java/org/carlspring/strongbox/providers/repository/RepositoryProvider.java

```diff
@@ -34,5 +37,13 @@ ArtifactOutputStream getOutputStream(String storageId,
     List<Path> search(RepositorySearchRequest searchRequest, RepositoryPageRequest pageRequest);
     
     Long count(RepositorySearchRequest searchRequest);
+    
+    Path getPath(String storageId,
+                 String repositoryId,
+                 String artifactPath)
+           throws IOException,
+                  NoSuchAlgorithmException,
+                  ArtifactTransportException,
+                  ProviderImplementationException;
```

Comment:
`
throws IOException,
NoSuchAlgorithmException,
ArtifactTransportException,
ProviderImplementationException;
`
I strongly believe it's too much just for getting path from repository provider.